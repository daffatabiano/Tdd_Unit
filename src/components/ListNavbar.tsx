import Link from 'next/link';

export default function ListNavbar() {
    return (
        <>
            <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                <li className="nav-item" role="presentation">
                    <Link
                        className="nav-link active"
                        aria-current="page"
                        href={'/'}
                    >
                        Home
                    </Link>
                </li>
                <li className="nav-item">
                    <Link className="nav-link" href={'/About'}>
                        About
                    </Link>
                </li>
                <li className="nav-item">
                    <Link
                        href={''}
                        className="nav-link disabled"
                        aria-disabled="true"
                    >
                        Disabled
                    </Link>
                </li>
            </ul>
        </>
    );
}
