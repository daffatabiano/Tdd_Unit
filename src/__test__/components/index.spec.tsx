import React from 'react';
import { render } from '@testing-library/react';
import ListNavbar from '../../components/ListNavbar';

describe('List component', () => {
    it('should render ', () => {
        const { container } = render(<ListNavbar />);
        expect(container).toMatchSnapshot();
    });
});

describe('List component', () => {
    it('should render text', () => {
        const { getByText } = render(<ListNavbar />);
        expect(getByText('Disabled')).toBeInTheDocument();
    });
});

describe('List component', () => {
    it('should render text', () => {
        const { getByRole } = render(<ListNavbar />);
        expect(getByRole('presentation')).toBeInTheDocument();
    });
});
