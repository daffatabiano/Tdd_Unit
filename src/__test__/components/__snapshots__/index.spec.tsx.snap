// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render  1`] = `
<div>
  <ul
    class="navbar-nav me-auto mb-2 mb-lg-0"
  >
    <li
      class="nav-item"
      role="presentation"
    >
      <a
        aria-current="page"
        class="nav-link active"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/About"
      >
        About
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        aria-disabled="true"
        class="nav-link disabled"
        href=""
      >
        Disabled
      </a>
    </li>
  </ul>
</div>
`;
